src/main/java/io/phdata/streamliner/schemadefiner/model/ColumnDefinition.java

```diff
@@ -35,4 +39,62 @@ public ColumnDefinition(String sourceName, String destinationName, String dataTy
         this.dataType = dataType;
         this.comment = comment;
     }
+
+    private String cleanseDataType(String dataType){
+        if(!dataType.endsWith(" IDENTITY")){
+            return dataType.toUpperCase();
+        }else{
+            int index = dataType.lastIndexOf(" IDENTITY");
+            return dataType.substring(0, index);
+        }
+    }
+
+  public String mapDataTypeSnowflake(
+      scala.collection.immutable.Map<String, scala.collection.immutable.Map<String, String>>
+          typeMapping) {
+    Map<String, Map<String, String>> javaTypeMap = JavaHelper.convertScalaMapToJavaMap(typeMapping);
+    String cleanDataType = cleanseDataType(dataType);
+    Integer p = precision == null ? 0 : precision;
+    Integer s = scale == null ? 0 : scale;
+
+    if (cleanDataType.equalsIgnoreCase("NUMBER")) {
+      if (p == 0 && (s == -127 || s == 0)) {
```

Comment:
Is this line the reason the standard type mapping can't be used?

Reply:
I think so.

Reply:
Yep